src/util.h

```diff
@@ -163,10 +163,14 @@ SECP256K1_GNUC_EXT typedef unsigned __int128 uint128_t;
 /* Zero memory if flag == 1. Constant time. */
 static SECP256K1_INLINE void memczero(void *s, size_t len, int flag) {
     unsigned char *p;
-    unsigned char mask = -(unsigned char)flag;
+    /* Access flag with a volatile-qualified lvalue.
+       This prevents clang from figuring out (after inlining) that flag can
+       take only be 0 or 1, which leads to variable time code. */
+    volatile int *flagp = &flag;
```

Comment:
This works too but I find volatile function parameters very obscure. Some C++ people agree with me: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1152r0.html#parret

Also the placement of the comment is more natural with this version.

Reply:
Afaics the argument isn't super strong ("it leaks function implementation information to the caller"). But what about making flagp a `volatile int` instead of pointer, and copy flag into it? Otherwise flagp is declared as a pointer to `volatile int`, but does not in fact point to a `volatile int`.

Reply:
I changed it to your suggestion. (I didn't do this previously because I wrongly assumed it needs one more store).

>  Otherwise flagp is declared as a pointer to `volatile int`, but does not in fact point to a `volatile int`.
When working on this PR I learned the truth about `volatile`. While the C standards speaks about "volatile objects", what the authors meant and luckily what every compiler implemented are "volatile lvalues" [1]. In other words, being volatile is a property of an access to an object, not a property of the object itself. So both variants with and without pointer are equally good in practice but the one without pointer is simpler and indeed closer the letter of the standard.   

[1] Defect report 476. http://www.open-std.org/jtc1/sc22/wg14/www/docs/summary.htm#dr_476